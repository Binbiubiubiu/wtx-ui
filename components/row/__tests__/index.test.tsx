import React from 'react';
import { render } from 'enzyme';
import Row from '..';
import toJson from 'enzyme-to-json';

describe('Row', () => {
  it('should render Row components', () => {
    const wrapper = render(
      <>
        <Row
          justify="start"
          style={{ width: 230, background: 'rgba(255,255,255,.2)', marginBottom: 20 }}
        >
          <div className="blue">flex-start</div>
          <div className="red"></div>
        </Row>
        <Row
          justify="center"
          style={{ width: 230, background: 'rgba(255,255,255,.2)', marginBottom: 20 }}
        >
          <div className="blue">center</div>
          <div className="red"></div>
        </Row>
        <Row
          justify="end"
          style={{ width: 230, background: 'rgba(255,255,255,.2)', marginBottom: 20 }}
        >
          <div className="blue">flex-end</div>
          <div className="red"></div>
        </Row>
        <Row
          justify="space-between"
          style={{ width: 230, background: 'rgba(255,255,255,.2)', marginBottom: 20 }}
        >
          <div className="blue">space-between</div>
          <div className="red"></div>
        </Row>
        <Row
          justify="space-around"
          style={{ width: 230, background: 'rgba(255,255,255,.2)', marginBottom: 20 }}
        >
          <div className="blue">space-around</div>
          <div className="red"></div>
        </Row>
      </>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should style output is not null', () => {
    expect(require('../style')).not.toBeNull();
  });
});
