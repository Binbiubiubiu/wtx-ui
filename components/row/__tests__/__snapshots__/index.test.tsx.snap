// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render Row components 1`] = `
Array [
  <div
    class="wtx-row wtx-row-flex-top wtx-row-flex-start"
    style="width:230px;background:rgba(255,255,255,.2);margin-bottom:20px"
  >
    <div
      class="blue"
    >
      flex-start
    </div>
    <div
      class="red"
    />
  </div>,
  <div
    class="wtx-row wtx-row-flex-top wtx-row-flex-center"
    style="width:230px;background:rgba(255,255,255,.2);margin-bottom:20px"
  >
    <div
      class="blue"
    >
      center
    </div>
    <div
      class="red"
    />
  </div>,
  <div
    class="wtx-row wtx-row-flex-top wtx-row-flex-end"
    style="width:230px;background:rgba(255,255,255,.2);margin-bottom:20px"
  >
    <div
      class="blue"
    >
      flex-end
    </div>
    <div
      class="red"
    />
  </div>,
  <div
    class="wtx-row wtx-row-flex-top wtx-row-flex-space-between"
    style="width:230px;background:rgba(255,255,255,.2);margin-bottom:20px"
  >
    <div
      class="blue"
    >
      space-between
    </div>
    <div
      class="red"
    />
  </div>,
  <div
    class="wtx-row wtx-row-flex-top wtx-row-flex-space-around"
    style="width:230px;background:rgba(255,255,255,.2);margin-bottom:20px"
  >
    <div
      class="blue"
    >
      space-around
    </div>
    <div
      class="red"
    />
  </div>,
]
`;
